src/app/donation-start/donation-start.component.ts

```diff
@@ -284,10 +295,14 @@ export class DonationStartComponent implements AfterContentChecked, OnDestroy, O
     const activelySelectedNext = (event.previouslySelectedStep.label === 'Your donation'
                                   && event.previouslySelectedStep.interacted === true);
 
-
-    // Only create donation if user actively clicks 'next' from first step, and previous donation is invalid.
+    // Proceed if user actively clicks 'next' from first step, and previous donation is invalid.
     if (activelySelectedNext && (this.previousDonation === undefined || this.previousDonation.status === 'Cancelled')) {
-      this.createDonation();
+
+      // Create donation if one hasn't been defined yet or
+      // if the original donation amount was updated.
+      if (this.donation === undefined || (this.donation && this.donationAmount !== this.donation.donationAmount)) {
```

Comment:
If the cancellation works we should be able to avoid having a separate check. Having 2 overlapping async things seems like asking for trouble with timing issues – we ought to be able to do the checks in 1 place.